logic/wireguard.go

```diff
@@ -346,9 +346,6 @@ func setWGKeyConfig(node *models.Node) error {
 }
 
 func removeLocalServer(node *models.Node) error {
-	if !IsLocalServer(node) {
```

Comment:
node is deleted from database prior to call to removeLocalServer and thus !IsLocalServer will always be true.